nntrainer/models/neuralnet.cpp

```diff
@@ -792,6 +796,10 @@ int NeuralNetwork::train_run() {
                         update_eval_stat, eval_epoch_end);
   }
 
+  /** Clear the set inputs and labels */
+  setLabels({}, model_graph.getLabelList());
+  setInputs({}, model_graph.getInputList());
```

Comment:
Wonder if we have to clear?

Reply:
is this deallocated input,label?

Reply:
The purpose is to unset the tensors which are not valid anymore as the dataset can free them. This is for maintaining the semantics.

Reply:
Okay